lib/src/repo.rs

```diff
@@ -160,15 +159,16 @@ impl ReadonlyRepo {
         fs::create_dir(repo_path.join("index")).unwrap();
     }
 
-    fn init(
+    pub fn init(
         user_settings: &UserSettings,
         repo_path: PathBuf,
-        store: Arc<Store>,
+        backend_factory: &dyn Fn(PathBuf) -> Box<dyn Backend>,
```

Comment:
Just out of curiosity. Is it for binary size optimization that we use `&dyn Fn` instead of `impl FnOnce`?

Reply:
No, I just had not reflected on which to use, so thanks for the tip. I've changed it to `impl FnOnce` now.